x/pairing/types/relay_exchange.go

```diff
@@ -0,0 +1,70 @@
+package types
+
+import (
+	"strings"
+
+	"github.com/lavanet/lava/utils"
+	"github.com/lavanet/lava/utils/sigs"
+)
+
+// RelayExchange consists a relay request and its corresponding response
+type RelayExchange struct {
+	Request RelayRequest
+	Reply   RelayReply
+}
+
+func NewRelayExchange(req RelayRequest, res RelayReply) RelayExchange {
+	return RelayExchange{Request: req, Reply: res}
+}
+
+func (re RelayExchange) GetSignature() []byte {
+	return re.Reply.Sig
+}
+
+func (re RelayExchange) DataToSign() []byte {
+	re.Reply.Sig = nil
+	var metadataBytes []byte
+	for _, metadata := range re.Reply.GetMetadata() {
+		data, err := metadata.Marshal()
+		if err != nil {
+			utils.LavaFormatError("metadata can't be marshaled to bytes", err)
+		}
+		metadataBytes = append(metadataBytes, data...)
+	}
+	// we remove the salt from the signature because it can be different
+	re.Request.RelayData.Salt = nil
+	msgParts := [][]byte{
+		re.Reply.GetData(),
+		[]byte(re.Request.RelayData.String()),
+		metadataBytes,
+	}
+	var totalLen int
```

Comment:
why is this?

Reply:
removed. see https://github.com/lavanet/lava/commit/d672a59e25d4d0a6ead4b6607716914718f99aab